neqo-common/src/qlog.rs

```diff
@@ -97,19 +97,6 @@ impl Qlog {
         Self::default()
     }
 
-    /// If logging enabled, closure may generate an event to be logged.
-    pub fn add_event_with_instant<F>(&mut self, f: F, now: Instant)
-    where
-        F: FnOnce() -> Option<qlog::events::Event>,
-    {
-        self.add_event_with_stream(|s| {
-            if let Some(evt) = f() {
-                s.add_event_with_instant(evt, now)?;
-            }
-            Ok(())
-        });
-    }
-
     /// If logging enabled, closure may generate an event to be logged.
     pub fn add_event_data_with_instant<F>(&mut self, f: F, now: Instant)
```

Comment:
Given that we don't have an `add_event()` or any other variant, I think that you want to name this `add_event_at()` and the one below `add_event()`.  I think that users will be able to handle the difference and being more succinct is good.